Documentation/ceph-disaster-recovery.md

```diff
@@ -336,3 +336,140 @@ Assuming `dataHostPathData` is `/var/lib/rook`, and the `CephCluster` trying to
 1. Bring the Rook Ceph operator back online by running `kubectl -n rook-ceph edit deploy/rook-ceph-operator` and set `replicas` to `1`.
 1. Watch the operator logs with `kubectl -n rook-ceph logs -f rook-ceph-operator-xxxxxxx`, and wait until the orchestration has settled.
 1. **STATE**: Now the new cluster should be up and running with authentication enabled. `ceph -s` output should not change much comparing to previous steps.
+
+
+# Adopt an existing Rook Ceph cluster (on PVs) into a new Kubernetes cluster
+
+> This section was written for a cluster running in a dynamic environment where the MONs and OSDs were running on PVs.
+> For example, the cluster may have been configured from [cluster-on-pvc.yaml](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/cluster-on-pvc.yaml).
+
+## Scenario
+
+1. The Kubernetes environment underlying a running Rook Ceph cluster failed catastrophically, requiring a new Kubernetes environment in which the user wishes to recover the previous Rook Ceph cluster.
+2. The underlying PVs with the Ceph data (OSDs) and metadata (MONs) are still available in the cloud environment.
+
+### Prerequisites
+
+1. A working Kubernetes cluster to which we will migrate the previous Rook Ceph cluster.
+2. Before the disaster, at least one Ceph mon db was in quorum, and a sufficient number of Ceph OSDs were `up` and `in`.
+3. The required resources were exported from the previous K8s cluster before it went down.
+
+### Exporting Critical Info
+
+Critical keys and info about the mons must be exported from the original cluster. This info is not stored on the PVs by either the mons
+or osds. This info is necessary to restore the cluster in case of disaster.
+
+```
+mkdir critical
+export namespace=rook-ceph
+kubectl -n ${namespace} get secret rook-ceph-mon -o yaml > critical/rook-ceph-mon.yaml
+kubectl -n ${namespace} get cm rook-ceph-mon-endpoints -o yaml > critical/rook-ceph-mon-endpoints.yaml
+kubectl -n ${namespace} get svc -l app=rook-ceph-mon -o yaml > critical/rook-ceph-mon-svc.yaml
+# information about PVCs and PVs to help reconstruct them later
+# TODO: Can we just export these as yamls and import them again directly? At a minimum we would need to filter the PV list since more than Rook PVs would be included.
+kubectl -n ${namespace} get pv -o yaml > critical/pvs.txt
+kubectl -n ${namespace} get pvc -o yaml > critical/pvcs.txt
+```
+
+### Restoring the Cluster
+
+1. Start the new Kubernetes clusterr
+
+2. Modify the critical resources before creating them
```

Comment:
Fields to trim:

* `creationTimestamp`
* `namespace` if different
* `resourceVersion`
* `uid`